diff --git a/src/plugin/plugin.cpp b/src/plugin/plugin.cpp
index aeb4600..3f44ef4 100644
--- a/src/plugin/plugin.cpp
+++ b/src/plugin/plugin.cpp
@@ -74,9 +74,9 @@ class Q_DECL_EXPORT NemoDeviceLockPlugin : public QQmlExtensionPlugin
     {
         qDBusRegisterMetaType<NemoDeviceLock::Fingerprint>();
         qDBusRegisterMetaType<QVector<NemoDeviceLock::Fingerprint>>();
-
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
         qmlRegisterType<NemoDeviceLock::FingerprintModel>();
-
+#endif
         qmlRegisterSingletonType<NemoDeviceLock::DeviceLock>(uri, 1, 0, "DeviceLock", createDeviceLock);
 
         qmlRegisterType<NemoDeviceLock::Authenticator>(uri, 1, 0, "Authenticator");
